File: web-frontend/src/api/translation.ts
import { apiClient, unwrapApiResponse } from './client';

import type {
  ApiResponse,
  CheckRequest,
  CheckResult,
  CompareTranslationResult,
  HealthResponse,
  TranslationRequest,
  TranslationResult,
} from '../types/api';

export async function healthCheck(): Promise<HealthResponse> {
  const response = await apiClient.get<HealthResponse>('/api/health');
  return response.data;
}

export async function translateWithKb(payload: TranslationRequest): Promise<TranslationResult> {
  const response = await apiClient.post<ApiResponse<TranslationResult>>('/api/v1/translate/kb', payload);
  return unwrapApiResponse(response.data);
}

export async function translateDirect(payload: TranslationRequest): Promise<TranslationResult> {
  const response = await apiClient.post<ApiResponse<TranslationResult>>('/api/v1/translate/direct', payload);
  return unwrapApiResponse(response.data);
}

export async function compareTranslate(payload: TranslationRequest): Promise<CompareTranslationResult> {
  const response = await apiClient.post<ApiResponse<CompareTranslationResult>>('/api/v1/translate/compare', payload);
  return unwrapApiResponse(response.data);
}

export async function checkTranslation(payload: CheckRequest): Promise<CheckResult> {
  const response = await apiClient.post<ApiResponse<CheckResult>>('/api/v1/check', payload);
  return unwrapApiResponse(response.data);
}

File: backend/app/api/routes/translation.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from app.core.config import Settings, get_settings
from app.schemas.api import ApiResponse
from app.schemas.translation import (
    CheckRequest,
    CheckResult,
    CompareTranslationResult,
    TranslationRequest,
    TranslationResult,
)
from app.services.check_service import CheckService
from app.services.dify_client import DifyClient
from app.services.translation_service import TranslationService


router = APIRouter(prefix="/api/v1", tags=["translation"])


def get_dify_client(settings: Settings = Depends(get_settings)) -> DifyClient:
    return DifyClient(settings)


def get_translation_service(
    settings: Settings = Depends(get_settings),
    dify_client: DifyClient = Depends(get_dify_client),
) -> TranslationService:
    return TranslationService(settings=settings, dify_client=dify_client)


def get_check_service(
    settings: Settings = Depends(get_settings),
    dify_client: DifyClient = Depends(get_dify_client),
) -> CheckService:
    return CheckService(settings=settings, dify_client=dify_client)


@router.post("/translate/kb", response_model=ApiResponse[TranslationResult])
async def translate_with_kb(
    payload: TranslationRequest,
    service: TranslationService = Depends(get_translation_service),
) -> ApiResponse[TranslationResult]:
    return ApiResponse(success=True, data=await service.translate_with_kb(payload), error=None)


@router.post("/translate/direct", response_model=ApiResponse[TranslationResult])
async def translate_direct(
    payload: TranslationRequest,
    service: TranslationService = Depends(get_translation_service),
) -> ApiResponse[TranslationResult]:
    return ApiResponse(success=True, data=await service.translate_direct(payload), error=None)


@router.post("/translate/compare", response_model=ApiResponse[CompareTranslationResult])
async def compare_translate(
    payload: TranslationRequest,
    service: TranslationService = Depends(get_translation_service),
) -> ApiResponse[CompareTranslationResult]:
    return ApiResponse(success=True, data=await service.compare_translate(payload), error=None)


@router.post("/check", response_model=ApiResponse[CheckResult])
async def check_translation(
    payload: CheckRequest,
    service: CheckService = Depends(get_check_service),
) -> ApiResponse[CheckResult]:
    return ApiResponse(success=True, data=await service.check_translation(payload), error=None)
